pkg/backingstore/reconciler.go

```diff
@@ -920,32 +927,56 @@ func (r *Reconciler) ReconcilePool() error {
 
 func (r *Reconciler) reconcilePvPool() error {
 	podsList := &corev1.PodList{}
+	pvcsList := &corev1.PersistentVolumeClaimList{}
 	util.KubeList(podsList, client.InNamespace(options.Namespace), client.MatchingLabels{"pool": r.BackingStore.Name})
+	util.KubeList(pvcsList, client.InNamespace(options.Namespace), client.MatchingLabels{"pool": r.BackingStore.Name})
+	if len(podsList.Items) < len(pvcsList.Items) {
+		r.reconcileDeletedPods(podsList, pvcsList)
```

Comment:
there is pod creation in both `reconcileDeletedPods` and `reconcileMissingPods`, and it's also not so clear from the name what's the purpose of each function.
I suggest making the flow clearer by ordering it differently: 
1. create PVCs
2. create pods
3. reconcile existing pods

something like that:
```go
func (r *Reconciler) reconcilePvPool() error {
	podsList := &corev1.PodList{}
	pvcsList := &corev1.PersistentVolumeClaimList{}
	util.KubeList(podsList, client.InNamespace(options.Namespace), client.MatchingLabels{"pool": r.BackingStore.Name})
	util.KubeList(pvcsList, client.InNamespace(options.Namespace), client.MatchingLabels{"pool": r.BackingStore.Name})

	if len(pvcsList.Items) < r.BackingStore.Spec.PVPool.NumVolumes {
                // only create new PVCs and not pods
		r.reconcilePVCS(podsList)
	}

        util.KubeList(pvcsList, client.InNamespace(options.Namespace), client.MatchingLabels{"pool": r.BackingStore.Name})
	if len(podsList.Items) < len(pvcsList.Items) {
                // create pods for orphan PVCs
		r.reconcileMissingPods(podsList, pvcsList)
	}

	return r.reconcileExistingPods(podsList)
}

```